cbrkit/sim/collections.py

```diff
@@ -190,3 +192,227 @@ def wrapped_func(query: Sequence[ValueType], case: Sequence[ValueType]) -> float
+
+
+@dataclass
+class ListMappingSim:
+    value: float
+    local_similarities: List[float] = None
+
+
+def list_mapping(
+    element_similarity: Callable[[Any, Any], float] = None,
+    exact: bool = False,
+    return_local_similarities: bool = False,
+) -> Callable[[Collection[Any], Collection[Any]], ListMappingSim]:
+    """List Mapping similarity function.
+
+    Parameters:
+    element_similarity: The similarity function to use for comparing elements.
+    exact: Whether to use exact or inexact comparison. Default is False (inexact).
+    return_local_similarities: Whether to return local similarities. Default is False.
+
+    Examples:
+        >>> sim = list_mapping(lambda x, y: 1.0 if x == y else 0.0, True, True)
+        >>> result = sim(["a", "b", "c"], ["a", "b", "c"])
+        >>> result.value
+        1.0
+        >>> result.local_similarities
+        [1.0, 1.0, 1.0]
+    """
+
+    def compute_contains_exact(
+        list1: Collection[Any], list2: Collection[Any]
+    ) -> ListMappingSim:
+        if len(list1) != len(list2):
+            return ListMappingSim(value=0.0)
+
+        sim_sum = 0.0
+        local_similarities = []
+
+        for elem1, elem2 in zip(list1, list2):
+            sim = element_similarity(elem1, elem2)
+            sim_sum += sim
+            local_similarities.append(sim)
+
+        if return_local_similarities:
+            return ListMappingSim(
+                value=sim_sum / len(list1), local_similarities=local_similarities
+            )
+        else:
+            return ListMappingSim(value=sim_sum / len(list1))
+
+    def compute_contains_inexact(
+        larger_list: Collection[Any], smaller_list: Collection[Any]
+    ) -> ListMappingSim:
+        max_similarity = -1.0
+        best_local_similarities = []
+
+        for i in range(len(larger_list) - len(smaller_list) + 1):
+            sublist = larger_list[i : i + len(smaller_list)]
+            sim_result = compute_contains_exact(sublist, smaller_list)
+
+            if sim_result.value > max_similarity:
+                max_similarity = sim_result.value
+                best_local_similarities = sim_result.local_similarities
+
+        if return_local_similarities:
+            return ListMappingSim(
+                value=max_similarity, local_similarities=best_local_similarities
+            )
+        else:
+            return ListMappingSim(value=max_similarity)
+
+    def wrapped_func(x: Collection[Any], y: Collection[Any]) -> ListMappingSim:
+        if exact:
+            return compute_contains_exact(x, y)
+        else:
+            if len(x) >= len(y):
+                return compute_contains_inexact(x, y)
+            else:
+                return compute_contains_inexact(y, x)
+
+    return wrapped_func
+
+
+@dataclass
+class Weight:
+    weight: float
+    lower_bound: float
+    upper_bound: float
+    inclusive_lower: bool
+    inclusive_upper: bool
+    normalized_weight: float = None  # Will be set later
```

Comment:
Type does not correspond to default value.

```suggestion
    normalized_weight: float | None = None  # Will be set later
```